### PR TITLE
Bug 2086092: bump to k8s v0.24.0

### PR DESCRIPTION
**Pull request by tkashem**

### CODE REVIEW

File: pkg/operator/configobserver/featuregates/observe_featuregates_test.go [hunk 1]
```diff
@@ -66,7 +66,6 @@ func TestObserveFeatureFlags(t *testing.T) {
 				"RotateKubeletServerCertificate=true",
 				"DownwardAPIHugePages=true",
 				"PodSecurity=true",
-				"CSIDriverAzureFile=true",
```
Comment (tkashem) [inline pkg/operator/configobserver/featuregates/observe_featuregates_test.go:69]: @soltysh @jsafrane can you check if this is the right fix for this test?

Comment (jsafrane) [reply pkg/operator/configobserver/featuregates/observe_featuregates_test.go:69]: Yes, the code is written in a way that the test must match openshift/api defaults [here](https://github.com/openshift/api/blob/441fe135b2fce3b4aac41a1488bea0b79cc6c800/config/v1/types_feature.go#L114)

(Some indirection would be nice though, feel free to fix it ;-) )

**Comment by tkashem** [comment]
> @tkashem do you want to update to 1.24 GA version and continue with this PR?

I think we will merge in this order:
- https://github.com/openshift/api/pull/1184
- https://github.com/openshift/client-go/pull/213
- https://github.com/openshift/library-go/pull/1356
- https://github.com/openshift/apiserver-library-go/pull/81

**Comment by tkashem** [comment]
good to merge

/hold cancel

**Review by ravisantoshgudimetla — APPROVED** [review]
/lgtm
/approve

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**